### PR TITLE
Use listing pattern throughout kubernetes package

### PR DESCRIPTION
**Pull request by stefwalter**
This adds listing behavior to the details page.

### CODE REVIEW

**Comment by stefwalter** [comment]
Once the filtering code lands in #2691 then we need to finish hooking this up to the filtering, and selection logic.

File: pkg/kubernetes/views/replicationcontroller-panel.html [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            <span ng-if="item.spec.replicas !== item.status.replicas">{{ item.status.replicas }} of {{ item.spec.replicas }}</span>
+        </div>
+        <h3 class="listing-title" ng-click="select(id)">{{ item.metadata.name }}</h3>
+    </div>
+    <tabset class="">
+        <tab heading="Replication Controller">
+            <div class="panel-body row">
+                <div class="col-sm-6">
+                    <dl>
+                        <dt>Namespace</dt>
```
Comment (petervo) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: translateable dt's

File: pkg/kubernetes/views/replicationcontroller-panel.html [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            </div>
+        </tab>
+        <tab heading="Pods">
+            <div class="panel-body row">
+                <div class="col-sm-6">
+                    <dl ng-init="pods = replicationcontrollerPods(item)">
+                        <dt class="header full" translatable="yes">Pod Replicated</dt>
+                        <dd class="full" ng-repeat="pod in pods">
+                            {{ pod.metadata.name }}</dd>
+                        <dd ng-if="!pods.count">No pods replicated</dd>
```
Comment (petervo) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: translatable

**Comment by petervo** [comment]
Should we grey out the delete and edit buttons till they are implemented, like the images listing.

**Comment by petervo** [comment]
The Templates tab and the container list seem to expand too much for me, causing horizontal scroll bars.

ss

File: pkg/kubernetes/views/container-body.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<div class="panel-body row">
+    <div class="col-sm-6">
+        <dl>
```
Comment (petervo) [inline pkg/kubernetes/views/container-body.html]: translatable dt's

File: pkg/kubernetes/views/container-body.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            <dd>{{state}}</dd>
+            <dt>Since</dt>
+            <dd ng-repeat-end>{{args.startedAt}}</dd>
+            <dt ng-if="container.status">Restart Count</dt>
+            <dd ng-if="container.status">{{container.status.restartCount}}</dd>
+        </dl>
+    </div>
+    <div class="col-sm-6">
+        <dl>
+            <dt class="header full" ng-if="container.spec.env.length">Environment</dt>
```
Comment (petervo) [inline pkg/kubernetes/views/container-body.html]: translatable

File: pkg/kubernetes/views/pod-body.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<div class="panel-body row">
+    <div class="col-sm-6">
+        <dl>
```
Comment (petervo) [inline pkg/kubernetes/views/pod-body.html]: translatable dt's

File: pkg/kubernetes/views/replicationcontroller-panel.html [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                    </dl>
+                </div>
+                <div class="col-sm-6">
+                    <dl>
+                        <dt class="header full" translatable="yes">Pod Selector</dt>
```
Comment (petervo) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: translatable

File: pkg/kubernetes/views/service-panel.html [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            <span ng-if="endpoint.subsets.length !== 0" translatable="yes">Ready</span>
+            <span ng-if="endpoint.subsets.length === 0" translatable="yes">Not Ready</span>
+        </div>
+        <h3 class="listing-title" ng-click="select(id)">{{ item.metadata.name }}</h3>
+    </div>
+    <tabset class="">
+        <tab heading="Service">
+            <div class="panel-body row">
+                <div class="col-sm-6">
+                    <dl>
```
Comment (petervo) [inline pkg/kubernetes/views/service-panel.html]: translatable dt's

File: pkg/kubernetes/views/container-body.html (hunk 3 above)
Comment (stefwalter) [inline pkg/kubernetes/views/container-body.html]: Fixed.

File: pkg/kubernetes/views/container-body.html (hunk 4 above)
Comment (stefwalter) [inline pkg/kubernetes/views/container-body.html]: Fixed.

File: pkg/kubernetes/views/pod-body.html (hunk 5 above)
Comment (stefwalter) [inline pkg/kubernetes/views/pod-body.html]: Fixed.

File: pkg/kubernetes/views/replicationcontroller-panel.html (hunk 1 above)
Comment (stefwalter) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: Fixed.

File: pkg/kubernetes/views/replicationcontroller-panel.html (hunk 2 above)
Comment (stefwalter) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: Fixed.

File: pkg/kubernetes/views/service-panel.html (hunk 7 above)
Comment (stefwalter) [inline pkg/kubernetes/views/service-panel.html]: Fixed.

**Comment by stefwalter** [comment]
> Should we grey out the delete and edit buttons till they are implemented, like the images listing.

Yes. Done.

> The Templates tab and the container list seem to expand too much for me, causing horizontal scroll bars.

I'd like to work on reflow and responsive design in another pull request, if that's okay with you?

**Pull request closed without merging**